File: packages/zodiac-dart/lib/src/models/chinese_year.dart
import 'chinese_animal.dart';
import 'zodiac_icon.dart';

/// Chinese Zodiac year result.
class ChineseYear {
  final ChineseAnimal animal;
  final String element;
  final String yinYang; // 'yin' | 'yang'
  final int cyclePosition;
  final int year;
  final String? locale;
  final String? localizedName;
  final String? localizedElement;
  final ZodiacIcon? icon;

  const ChineseYear({
    required this.animal,
    required this.element,
    required this.yinYang,
    required this.cyclePosition,
    required this.year,
    this.locale,
    this.localizedName,
    this.localizedElement,
    this.icon,
  });
}

File: packages/zodiac-js/src/models/chinese-year.ts
/**
 * Chinese Zodiac year result.
 *
 * Full result for a Chinese Zodiac year calculation including
 * animal, element, yin/yang, and 60-year cycle position.
 */
import type { ChineseAnimal } from './chinese-animal';
import type { ZodiacIcon } from './zodiac-icon';

export interface ChineseYear {
  /** The resolved Chinese Zodiac animal */
  readonly animal: ChineseAnimal;

  /** The heavenly stem element for this year */
  readonly element: string;

  /** Yin/Yang for this specific year */
  readonly yinYang: 'yin' | 'yang';

  /** Position in the 60-year sexagenary cycle (1–60) */
  readonly cyclePosition: number;

  /** The Gregorian year this result corresponds to */
  readonly year: number;

  /** Locale used for localized fields (if any) */
  readonly locale?: string;

  /** Localized animal name (if locale was provided) */
  readonly localizedName?: string;

  /** Localized element name (if locale was provided) */
  readonly localizedElement?: string;

  /** Resolved icon (if icon mode is enabled) */
  readonly icon?: ZodiacIcon;
}
